src/rpc/ngHandlers/LedgerEntry.cpp

```diff
@@ -0,0 +1,289 @@
+        return RPC::Status{
+            RPC::RippledError::rpcINVALID_PARAMS, "missingOwnerOrDirRoot"};
+
+    uint64_t const subIndex = directory.contains("sub_index")
+        ? boost::json::value_to<uint64_t>(directory.at("sub_index"))
+        : 0;
+
+    if (directory.contains("dir_root"))
+    {
+        ripple::uint256 const uDirRoot{
+            directory.at("dir_root").as_string().c_str()};
+        return ripple::keylet::page(uDirRoot, subIndex).key;
+    }
+
+    auto const ownerID = ripple::parseBase58<ripple::AccountID>(
+        directory.at("owner").as_string().c_str());
+    return ripple::keylet::page(ripple::keylet::ownerDir(*ownerID), subIndex)
+        .key;
+}
+
+void
+tag_invoke(
+    boost::json::value_from_tag,
+    boost::json::value& jv,
+    LedgerEntryHandler::Output const& output)
+{
+    auto object = boost::json::object{
+        {"ledger_hash", output.ledgerHash},
+        {"ledger_index", output.ledgerIndex},
+        {"validated", output.validated},
+        {"index", output.index}};
+    if (output.nodeBinary)
+    {
+        object["node_binary"] = *(output.nodeBinary);
+    }
+    else
+    {
+        object["node"] = *(output.node);
+    }
+    jv = std::move(object);
+}
+
+LedgerEntryHandler::Input
+tag_invoke(
+    boost::json::value_to_tag<LedgerEntryHandler::Input>,
+    boost::json::value const& jv)
+{
+    auto const& jsonObject = jv.as_object();
+    LedgerEntryHandler::Input input;
+    if (jsonObject.contains("ledger_hash"))
+    {
+        input.ledgerHash = jv.at("ledger_hash").as_string().c_str();
+    }
+    if (jsonObject.contains("ledger_index"))
+    {
+        if (!jsonObject.at("ledger_index").is_string())
+        {
+            input.ledgerIndex = jv.at("ledger_index").as_int64();
+        }
+        else if (jsonObject.at("ledger_index").as_string() != "validated")
+        {
+            input.ledgerIndex =
+                std::stoi(jv.at("ledger_index").as_string().c_str());
+        }
+    }
+    if (jsonObject.contains("binary"))
+    {
+        input.binary = jv.at("binary").as_bool();
+    }
+    // check all the protential index
+    static auto const indexFieldTypeMap =
+        std::unordered_map<std::string, ripple::LedgerEntryType>{
+            {"index", ripple::ltANY},
+            {"directory", ripple::ltDIR_NODE},
+            {"offer", ripple::ltOFFER},
+            {"check", ripple::ltCHECK},
+            {"escrow", ripple::ltESCROW},
+            {"payment_channel", ripple::ltPAYCHAN},
+            {"deposit_preauth", ripple::ltDEPOSIT_PREAUTH},
+            {"ticket", ripple::ltTICKET}};
+
+    auto const indexFieldType = std::find_if(
+        indexFieldTypeMap.begin(),
+        indexFieldTypeMap.end(),
+        [&jsonObject](auto const& pair) {
+            auto const& [field, _] = pair;
+            return jsonObject.contains(field) &&
+                jsonObject.at(field).is_string();
+        });
+    if (indexFieldType != indexFieldTypeMap.end())
```

Comment:
minor: I think this whole sheet of `if`s can be made into `if-elseif` with just one `return input` at the very bottom of the function. wdyt?